docs/graph_transformation.rst

```diff
@@ -0,0 +1,149 @@
+.. _graph_transformation:
+
+Graph Transformations
+=====================
+
+We introduce a series of method that can append more labels to a existed grpah, and
+do projection over existed graph. We will also show how to make a complex property graph
+compatible with algorithms that can only run on simple graph. Finally, we show how to add
+the query result of algorithm back to graph as a property on vertex.
+
+More specically, :class:`Graph` provides two methods for append labels, and one method for
+projection.
+
+
+.. code:: python
+
+    def add_vertices(self, vertices, label="_", properties=[], vid_field=0):
+        pass
+
+    def add_edges(self, edges, label="_", properties=[], src_label=None, dst_label=None, src_field=0, dst_field=1):
+        pass
+
+    def project(self, vertices, edges):
+        pass
+
+
+We have already seem `add_vertices` and `add_edges` in :ref:`loading graphs<loading_graphs>`, we use them
+to build a graph iteratively.
+
+Further, we can use them to attach more vertex labels and edge labels to a existed graph.
+But this won't modify the source graph, instead, it will return a new graph, which is based
+on the source graph.
+
+
+Attach new labels
+-----------------
+
+Take LDBC-SNB Property Graph as an example，We now load a subset of labels, as the source graph.
+
+
+.. code:: python
+
+    import graphscope
+    from pathlib import Path
+    from graphscope.framework.loader import Loader
+
+    sess = graphscope.session()
+
+    graph = sess.g(directed=directed)
```

Comment:
should I(the user) assign a `directed` variable before the statement ? because I don't know where the `directed` from in the example

Reply:
The `directed` here is not necessary, I copied it from the `ldbc.py` :). 
But the `sess.g()` and parameters is covered in `loading_graphs.rst`.